src/data_config.c

```diff
@@ -210,6 +210,9 @@ int data_config_pcre_exec(data_config *dc, cond_cache_t *cache, buffer *b) {
         cache->comp_value = b; /* holds pointer to b (!) for pattern subst */
     return cache->patterncount;
 #else
+    (void)dc;
```

Comment:
Compiling without warnings when PCRE is disabled (e.g. "./configure --without-pcre") should be a separate commit.  Multiple commits are permitted to be part of a single Pull Requests.  In any case, I will fix these separately.